src/compiler/checker.ts

```diff
@@ -9293,6 +9743,15 @@ namespace ts {
                     return anyType;
                 }
             }
+
+            let exprType = getTypeOfExpression(node.expression);

```

Comment:
This should use `checkExpression`. But a bigger issue is that this ends up calling `checkExpression` *twice* for every call (because it is also being called from `getResolvedSignature`). That is definitely something that needs fixing.